### PR TITLE
Migrate nightly benchmarks to GH Actions

### PR DESCRIPTION
**Pull request by reakaleek**
## Motivation/summary

Migrate nightly benchmarks from Jenkins to GH Actions

## How to test these changes

Tested here: https://github.com/elastic/apm-server/actions/runs/5412428491

### CODE REVIEW

File: testing/benchmark/Makefile [hunk 1]
```diff
@@ -20,15 +20,20 @@ GOBENCH_HOST ?= http://localhost:9200
...
+SSH_OPTS ?= -o LogLevel=ERROR -o StrictHostKeyChecking=no -o ServerAliveInterval=60 -o ServerAliveCountMax=10
 SSH_KEY ?= ~/.ssh/id_rsa_terraform
 WORKER_IP = $(shell terraform output -raw public_ip)
 
 SHELL = /bin/bash
 .SHELLFLAGS = -o pipefail -c
 
 # This profile will also be used by the Terraform provider.
-export AWS_PROFILE ?= default
+
```
Comment (reakaleek) [inline testing/benchmark/Makefile:31]: terraform was failing if `AWS_PROFILE` was defined but no corresponding profile could be found in config.

related issue: https://discuss.hashicorp.com/t/error-error-configuring-terraform-aws-provider-failed-to-get-shared-config-profile-default/39417/3

File: testing/benchmark/Makefile [hunk 2]
```diff
@@ -20,15 +20,20 @@ GOBENCH_HOST ?= http://localhost:9200
...
+# export AWS_PROFILE if CI is not defined
+ifeq ($(CI),)
+	export AWS_PROFILE ?= default
+endif
```
Comment (reakaleek) [inline testing/benchmark/Makefile:35]: terraform was failing if `AWS_PROFILE` was defined but no corresponding profile could be found in config.

related issue: https://discuss.hashicorp.com/t/error-error-configuring-terraform-aws-provider-failed-to-get-shared-config-profile-default/39417/3

**Review by kruskall — APPROVED** [review]
THANK YOU

File: .github/workflows/benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          vaultUrl: ${{ secrets.VAULT_ADDR }}
+          vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
+          vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
+          slackChannel: "#apm-sever"
+
+      - name: Post to a Slack channel
+        id: slack
+        uses: slackapi/slack-github-action@v1.24.0
+        with:
+          channel-id: "#apm-sever"
```
Comment (axw) [inline .github/workflows/benchmarks.yml:141]:
```suggestion
          channel-id: "#apm-server"
```

**Review by axw — APPROVED** [review]
:heart_eyes: ++ thank you!

Would it theoretically be possible for us to add `workflow_dispatch` later? I would love to be able to easily trigger the benchmarks off an arbitrary branch.

File: .github/workflows/benchmarks.yml (hunk 3 above)
Comment (reakaleek) [reply .github/workflows/benchmarks.yml:141]: Thank you for noticing this.

**Comment by reakaleek** [comment]
> 😍 ++ thank you!
> 
> Would it theoretically be possible for us to add `workflow_dispatch` later? I would love to be able to easily trigger the benchmarks off an arbitrary branch.

Yes, I was planning to do this in a follow up and discuss with you. But since you already mentioned `workflow_dispatch`. That will be the way to go.

**Pull request merged**